### DISCUSSION TITLE
frb v2.0.0-dev.38 generated reference to private module and get errors

### DISCUSSION THREAD

**Discussion by svr2kos2** [fzyzcjy/flutter_rust_bridge#2073] [Q&A] [answered]
Hi there, I'm newbie to rust. 
Here is my code:
```
/api
├── hid
│   ├── mod.rs
│   ├── hid_api.rs
│   ├── os_hid.rs
│   └── web_hid.rs
└── mod.rs

```
``` rust
// api/hid/mod.rs
pub mod hid_api;
```
``` rust
// api/hid/hid_api.rs
#[cfg(not(target_arch = "wasm32"))] 
#[path ="os_hid.rs"] mod platform_hid;
#[cfg(target_arch = "wasm32")] 
#[path ="web_hid.rs"] mod platform_hid;
```
``` rust
// api/mod.rs
pub mod hid;
```
In `v2.0.0-dev.37` it works fine.
But when I upgrade to `v2.0.0-dev.38` this error occurred
```
src\frb_generated.rs:24:27 
use crate::api::hid::hid_api::platform_hid::*;
                              ^^^^^^^^^^^^ private module
```
I tried making the following changes and it worked,  
but `platform_hid` doesn't need to expose any member to flutter.  
So I think it's not a right way to do it.
``` rust
// api/hid/hid_api.rs
#[cfg(not(target_arch = "wasm32"))] 
#[path ="os_hid.rs"] pub mod platform_hid;
#[cfg(target_arch = "wasm32")] 
#[path ="web_hid.rs"] pub mod platform_hid;
```
  
 Much thanks for any helps.

**Comment by fzyzcjy** [comment]
Looks like a bug due to refactor of dev.38 to support scanning third party crates. Could you please provide a minimal reproducible sample and create an issue, then I can fix that.

Without more info, I guess maybe it is because you have some function that points to this, or more possibly, do you have `pub` functions inside that mod? That may confuse it.
If that's the problem, I may update it to skip private mods.

EDIT: It worked before possibly because frb did not handle `#[path=...]`, thus the mod is as if nothing to old frb.

EDIT: Another way may be https://cjycode.com/flutter_rust_bridge/guides/miscellaneous/multi-input modify `rust_input` to only scan the folders you like.

**Reply by svr2kos2** [reply, marked as answer]
With a new flutter_rust_bridge_codegen project
add `api/internal_mod.rs`
``` rust
pub fn internal_hello() {
    println!("Hello from internal_mod.rs!");
}
```
add `api/external_mod.rs`
``` rust
#[path ="internal_mod.rs"] mod internal_hello;
pub fn external_hello() {
    println!("Hello from external_mod.rs!");
    internal_hello::internal_hello();
}
```
modify `api/mod.rs`
``` rust
pub mod simple;
pub mod external_mod;
```
`flutter_rust_bridge_codegen generate` and `build(web or windows)` then get

```
error[E0603]: module `internal_hello` is private
   --> src\frb_generated.rs:103:51
    |
103 |                         crate::api::external_mod::internal_hello::internal_hello();        
    |                                                   ^^^^^^^^^^^^^^  -------------- function `internal_hello` is not publicly re-exported
    |                                                   |
    |                                                   private module
```
As a beginner, I'm not sure if this is the wrong way to use internal code or a bug.  
But I found that I can use this way to avoid this by changing `internal_mod.rs` to:
``` rust
pub(in crate::api) fn internal_hello() {
    println!("Hello from internal_mod.rs!");
}
```

**Reply by fzyzcjy** [reply]
For future readers: https://github.com/fzyzcjy/flutter_rust_bridge/issues/2074
